libraries/app/application.cpp

```diff
@@ -537,7 +537,7 @@ namespace detail {
             _is_finished_syncing = true;
```

Comment:
Actually this code block can be removed because it will never execute?
Either already returned, or an exception was thrown out.